llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1447,6 +1484,388 @@ LogicalResult TopkOp::getResultTilePosition(
+}
+
+// Interchange `elements` starting at offset `offset` based on the indexes in
+// `interchangeVector`.
+template <typename T>
+static SmallVector<T> interchange(ArrayRef<T> elements,
+                                  ArrayRef<int64_t> interchangeVector,
+                                  int64_t offset) {
+  SmallVector<T> rearrangedElements = llvm::to_vector(elements);
+  if (interchangeVector.empty())
+    return rearrangedElements;
+  assert((rearrangedElements.size() - offset) == interchangeVector.size() &&
+         "number of elements must equal number of permutations");
+  for (int64_t idx = 0, end = interchangeVector.size(); idx < end; idx++)
+    rearrangedElements[interchangeVector[idx] + offset] =
+        elements[idx + offset];
+  return rearrangedElements;
+}
+
+// Infer result/output type given the input and the tile sizes.
+ShapedType PackOp::inferResultType() {
+  DenseMap<int64_t, int64_t> tileAndPosMapping = getDimAndStaticTileMapping();
+  SmallVector<int64_t> inferredShape;
+  inferredShape.reserve(getOutputRank());
+  ShapedType inputType = getInputType();
+  int64_t rank = getInputRank();
+
+  // tile loop.
+  for (auto i : llvm::seq<int64_t>(0, rank)) {
+    if (tileAndPosMapping.count(i)) {
+      if (inputType.isDynamicDim(i) ||
+          tileAndPosMapping[i] == ShapedType::kDynamicSize)
+        inferredShape.push_back(ShapedType::kDynamicSize);
+      else {
+        int64_t sizeDim =
+            ceilDiv(inputType.getDimSize(i), tileAndPosMapping[i]);
+        inferredShape.push_back(sizeDim);
+      }
+    } else
+      inferredShape.push_back(inputType.getShape()[i]);
+  }
+
+  // point loop.
+  SmallVector staticTiles = getStaticTiles();
+  inferredShape.append(staticTiles.begin(), staticTiles.end());
+
+  return TypeSwitch<Type, ShapedType>(inputType)
+      .Case<RankedTensorType>([&](RankedTensorType t) -> ShapedType {
+        return RankedTensorType::get(inferredShape, inputType.getElementType());
+      })
+      .Case<MemRefType>([&](MemRefType t) -> ShapedType {
+        return MemRefType::get(inferredShape, inputType.getElementType());
+      })
+      .Default([&](Type t) {
+        llvm_unreachable("unexpected type");
+        return nullptr;
+      });
+}
+
+// Return true if at least one element in `tiles` is zero.
+static bool hasZeros(ArrayRef<OpFoldResult> tiles) {
+
+  // Check if `tile` is an integer zero attribute or
+  // comes from a zero constantIndexOp.
+  auto isZero = [](OpFoldResult tile) -> bool {
+    if (Attribute attr = tile.dyn_cast<Attribute>()) {
+      IntegerAttr intAttr = attr.dyn_cast<IntegerAttr>();
+      return intAttr && intAttr.getValue().isZero();
+    } else if (auto cst =
+                   tile.get<Value>().getDefiningOp<arith::ConstantIndexOp>())
+      return cst.value() == 0;
+    return false;
+  };
+
+  return llvm::any_of(tiles, [&](OpFoldResult tile) { return isZero(tile); });
```

Comment:
You can just replace `isZero` with `isConstantIntValue` from `StaticValueUtils.h`